builtin/credential/aws/path_role.go

```diff
@@ -742,29 +811,81 @@ func (b *backend) pathRoleCreateUpdate(ctx context.Context, req *logical.Request
+	BoundIamRoleARN             string        `json:"bound_iam_role_arn,omitempty" structs:"bound_iam_role_arn" mapstructure:"bound_iam_role_arn"`
+	BoundIamRoleARNs            []string      `json:"bound_iam_role_arn_list" structs:"bound_iam_role_arn_list" mapstructure:"bound_iam_role_arn_list"`
+	BoundIamInstanceProfileARN  string        `json:"bound_iam_instance_profile_arn,omitempty" structs:"bound_iam_instance_profile_arn" mapstructure:"bound_iam_instance_profile_arn"`
+	BoundIamInstanceProfileARNs []string      `json:"bound_iam_instance_profile_arn_list" structs:"bound_iam_instance_profile_arn_list" mapstructure:"bound_iam_instance_profile_arn_list"`
+	BoundRegion                 string        `json:"bound_region,omitempty" structs:"bound_region" mapstructure:"bound_region"`
+	BoundRegions                []string      `json:"bound_region_list" structs:"bound_region_list" mapstructure:"bound_region_list"`
+	BoundSubnetID               string        `json:"bound_subnet_id,omitempty" structs:"bound_subnet_id" mapstructure:"bound_subnet_id"`
+	BoundSubnetIDs              []string      `json:"bound_subnet_id_list" structs:"bound_subnet_id_list" mapstructure:"bound_subnet_id_list"`
+	BoundVpcID                  string        `json:"bound_vpc_id,omitempty" structs:"bound_vpc_id" mapstructure:"bound_vpc_id"`
+	BoundVpcIDs                 []string      `json:"bound_vpc_id_list" structs:"bound_vpc_id_list" mapstructure:"bound_vpc_id_list"`
+	InferredEntityType          string        `json:"inferred_entity_type" structs:"inferred_entity_type" mapstructure:"inferred_entity_type"`
+	InferredAWSRegion           string        `json:"inferred_aws_region" structs:"inferred_aws_region" mapstructure:"inferred_aws_region"`
+	ResolveAWSUniqueIDs         bool          `json:"resolve_aws_unique_ids" structs:"resolve_aws_unique_ids" mapstructure:"resolve_aws_unique_ids"`
+	RoleTag                     string        `json:"role_tag" structs:"role_tag" mapstructure:"role_tag"`
+	AllowInstanceMigration      bool          `json:"allow_instance_migration" structs:"allow_instance_migration" mapstructure:"allow_instance_migration"`
+	TTL                         time.Duration `json:"ttl" structs:"ttl" mapstructure:"ttl"`
+	MaxTTL                      time.Duration `json:"max_ttl" structs:"max_ttl" mapstructure:"max_ttl"`
+	Policies                    []string      `json:"policies" structs:"policies" mapstructure:"policies"`
+	DisallowReauthentication    bool          `json:"disallow_reauthentication" structs:"disallow_reauthentication" mapstructure:"disallow_reauthentication"`
+	HMACKey                     string        `json:"hmac_key" structs:"hmac_key" mapstructure:"hmac_key"`
+	Period                      time.Duration `json:"period" mapstructure:"period" structs:"period"`
+	version                     int           `json:"version" mapstructure:"version" structs:"versoin"`
```

Comment:
Needs to be exported.

Reply:
Also, there is a typo in the `structs` tag.

Also, if structs isn't being used anymore, which it seems it isn't, and which is my preference, just remove the structs tags :-D

Reply:
done, including removing the structs tags entirely from this type.

Honestly, was never 100% sure why `structs` and `mapstructure` tags were included and I just kept including them because they had been included. If they're not needed, I'm also 100% in favor of removing them :)